build-tools/packages/build-tools/src/repoPolicyCheck/handlers/noJsFiles.ts

```diff
@@ -0,0 +1,19 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Handler } from "../common";
+
+/**
+ * A policy handler that checks for JavaScript source files that just use the .js file extension. Such files may be
+ * interpreted by node as either CommonJS or ESM based on the `type` field in the nearest package.json file. However
+ * this is not deterministic. Using file extensions reduces ambiguity and makes the behavior deterministic.
```

Comment:
What is not deterministic? You just mean that _we_ might change the package type?

Reply:
Doesn't seem not related. That is the system.
Brings to mind the missing fluid build dependency on package.json. Change package.json as you like. Build cache doesn't care.

Reply:
Clarified the comment. I understand that this is node's behavior, but it is not obvious, and it's also new(ish).